cpp/src/arrow/engine/substrait/options.h

```diff
@@ -93,6 +114,10 @@ struct ARROW_ENGINE_EXPORT ConversionOptions {
   /// The default behavior will return an invalid status if the plan has any
   /// named table relations.
   NamedTableProvider named_table_provider = kDefaultNamedTableProvider;
+  /// \brief A custom strategy to be used for mapping a tap kind to a function name
+  ///
+  /// The default mapper returns a declaration whose factory name is equal to the tap kind
+  NamedTapKindMapper named_tap_mapper = kDefaultNamedTapKindMapper;
```

Comment:
How do you feel about renaming this to `NamedTapProvider` for consistency?

Reply:
Done.